In [1]:
import pandas as pd
import requests
import zipfile
import io

url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/00357/occupancy_data.zip'
response = requests.get(url)

byte_data = io.BytesIO(response.content) # mimicks file placement in hard drive (virtual file)

zip_folder = zipfile.ZipFile(byte_data)

print(zip_folder.namelist())

['datatest.txt', 'datatest2.txt', 'datatraining.txt']


In [2]:
df = pd.read_csv(zip_folder.open('datatraining.txt'), sep=',')

print(len(df))
df.columns = [x.lower().strip() for x in df.columns] #clean column names
df.head()

8143


,date,temperature,humidity,light,co2,humidityratio,occupancy
1,2015-02-04 17:51:00,23.18,27.2720,426.0,721.25,0.004793,1
2,2015-02-04 17:51:59,23.15,27.2675,429.5,714.00,0.004783,1
3,2015-02-04 17:53:00,23.15,27.2450,426.0,713.50,0.004779,1
4,2015-02-04 17:54:00,23.15,27.2000,426.0,708.25,0.004772,1
5,2015-02-04 17:55:00,23.10,27.2000,426.0,704.50,0.004757,1


## <u>***Correlation of Occupancy with Light, Temperature and CO2***</u>

In [3]:
import scipy.stats as stats

In [13]:
corr_l, pval_l = stats.pearsonr(df['light'], df['occupancy']) # correlation between light and occupancy
corr_t, pval_t = stats.pearsonr(df['temperature'], df['occupancy']) # correlation between temperature and occupancy
corr_c, pval_c = stats.pearsonr(df['co2'], df['occupancy']) # correlation between co2 and occupancy

print(f"Correlation between light and occupancy: {corr_l:.2f}")
print(f"Correlation between temperature and occupancy: {corr_t:.2f}")
print(f"Correlation between CO2 and occupancy: {corr_c:.2f}")

Correlation between light and occupancy: 0.91
Correlation between temperature and occupancy: 0.54
Correlation between CO2 and occupancy: 0.71


##### The correlation analysis reveals that: 

1. **lighting state** is the most reliable indicator of room occupancy ***(r = 0.91)***. This aligns with expected occupant behavior, as lighting is an immediate, manual response to entering a space. 

2. **CO2 concentration** serves as a strong secondary metric ***(r = 0.71)***, accurately reflecting human metabolic load, though it is subject to the room's ventilation rate. 

3. **Temperature** showed the weakest relationship to immediate occupancy ***(r = 0.54)***, likely due to the thermal mass of the building and HVAC system lag.